### DISCUSSION TITLE
Public API to get Mastodon Servers info

### DISCUSSION THREAD

**Discussion by 0GiS0** [mastodon/mastodon#21578] [General]
Hi folks!

I'm building a website with mastodon servers info: https://0gis0.github.io/mastodon-servers-info/

I'm using this API: https://api.joinmastodon.org/servers but It seems the info is not updated. Some servers are missing and the total of users is outdated.

Do you know if there is a way to get this info or another API?

Thanks!!

**Comment by rbairwell** [comment]
I think that server list only includes servers which have a) signed up to the [Mastodon server covenant](https://joinmastodon.org/covenant) AND have open registrations.

You might find the list on https://instances.social/api/doc/#api-Instances-ListInstances more useful as that is a more generic orientated server list.

**Reply by KaKi87** [reply]
Hello,

Is there any alternative that does not require authentication ?

Thanks

**Comment by shaedrich** [comment]
You could call `/.well-known/nodeinfo` which will respond with something like this:
```json
{
  "links": [
    {
      "rel": "http://nodeinfo.diaspora.software/ns/schema/2.0",
      "href": "https://mastodon.online/nodeinfo/2.0"
    }
  ]
}
```

if you call `/nodeinfo/2.0` from `$.links[0].href`, you'll get in return something like
```json
{
  "version": "2.0",
  "software": {
    "name": "mastodon",
    "version": "4.3.0-nightly.2024-08-30"
  },
  "protocols": [
    "activitypub"
  ],
  "services": {
    "outbound": [],
    "inbound": []
  },
  "usage": {
    "users": {
      "total": 190561,
      "activeMonth": 13239,
      "activeHalfyear": 24093
    },
    "localPosts": 9599337
  },
  "openRegistrations": false,
  "metadata": {
    "nodeName": "Mastodon",
    "nodeDescription": "A newer server operated by the Mastodon gGmbH non-profit"
  }
}
```
